### PR TITLE
Adapted to class structure style policy - Part 3

### PR DESCRIPTION
**Pull request by thomasfanell**
This PR covers:
* source/geometry

### CODE REVIEW

File: source/geometry/FloatEuclidTransform.ooc [hunk 1]
```diff
@@ -63,8 +67,4 @@ FloatEuclidTransform: cover {
 		}
 		result
 	}
-	toString: func -> String {
```
Comment (marcusnaslund) [inline source/geometry/FloatEuclidTransform.ooc]: Why did toString move? It's not static.

File: source/geometry/FloatTransform2D.ooc [hunk 2]
```diff
@@ -172,9 +132,49 @@ FloatTransform2D: cover {
...
+					case 2 => result = this f
+				}
+			case 2 =>
+				match (y) {
+					case 0 => result = this g
+					case 1 => result = this h
+					case 2 => result = this i
+				}
+		}
+		result
```
Comment (marcusnaslund) [inline source/geometry/FloatTransform2D.ooc]: Is there any way to write this ugly piece of code in a nicer way?

File: source/geometry/FloatEuclidTransform.ooc (hunk 1 above)
Comment (thomasfanell) [inline source/geometry/FloatEuclidTransform.ooc]: It was probably below the operators?

Comment (marcusnaslund) [inline source/geometry/FloatEuclidTransform.ooc]: Oh nvm, I was looking at the wrong side. Time for vacation.

**Comment by marcusnaslund** [comment]
You could also look at reordering the operators themselves, to group similar lines together. That would make for more clean code. For example:
```
	operator + (other: This) -> This { This new(this x + other x, this y + other y, this z + other z) }
	operator + (other: IntVector3D) -> This { This new(this x + other x, this y + other y, this z + other z) }
	operator - (other: This) -> This { This new(this x - other x, this y - other y, this z - other z) }
	operator - (other: IntVector3D) -> This { This new(this x - other x, this y - other y, this z - other z) }
	operator - -> This { This new(-this x, -this y, -this z) }
	operator * (other: This) -> This { This new(this x * other x, this y * other y, this z * other z) }
	operator * (other: IntVector3D) -> This { This new(this x * other x, this y * other y, this z * other z) }
	operator / (other: This) -> This { This new(this x / other x, this y / other y, this z / other z) }
	operator / (other: IntVector3D) -> This { This new(this x / other x, this y / other y, this z / other z) }
```
looks a lot better as
```
	operator + (other: This) -> This { This new(this x + other x, this y + other y, this z + other z) }
	operator - (other: This) -> This { This new(this x - other x, this y - other y, this z - other z) }
	operator * (other: This) -> This { This new(this x * other x, this y * other y, this z * other z) }
	operator / (other: This) -> This { This new(this x / other x, this y / other y, this z / other z) }

	operator + (other: IntVector3D) -> This { This new(this x + other x, this y + other y, this z + other z) }
	operator - (other: IntVector3D) -> This { This new(this x - other x, this y - other y, this z - other z) }
	operator * (other: IntVector3D) -> This { This new(this x * other x, this y * other y, this z * other z) }
	operator / (other: IntVector3D) -> This { This new(this x / other x, this y / other y, this z / other z) }
	operator - -> This { This new(-this x, -this y, -this z) }
```

**Comment by marcusnaslund** [comment]
Above code is from `IntPoint3D` btw

**Comment by thomasfanell** [comment]
I agree, I should take this opportunity to organize the operators. The scrambled order bothers me a bit.

**Comment by marcusnaslund** [comment]
Awesome! :)

**Comment by thomasfanell** [comment]
So we should separate the operators (by type) with one blank line?

**Comment by thomasfanell** [comment]
Actually, I don't think we should separate by a blank line. It looks ugly when we mix oneliners with larger operator bodies.

**Comment by marcusnaslund** [comment]
Not sure what you mean there, but if it helps readability with a blank line, go for it. Otherwise don't. But we're generally not that keen on blank lines. Also, remember what I said about compacting one-liners, it depends on the context.

File: source/geometry/FloatTransform2D.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/geometry/FloatTransform2D.ooc]: @thomasfanell

File: source/geometry/IntBox2D.ooc [hunk 3]
```diff
@@ -102,6 +102,8 @@ IntBox2D: cover {
 		this left <= point x && point x < this right && this top <= point y && point y < this bottom
 	}
 	contains: func ~box (box: This) -> Bool { this intersection(box) == box }
+	toFloatBox2D: func -> FloatBox2D { FloatBox2D new(this left, this top, this width, this height) }
```
Comment (marcusnaslund) [inline source/geometry/IntBox2D.ooc]: Is there an implicit cast `as` that you can overload for `FloatBox2D`?
like `floatbox := myIntBox as FloatBox2D`?

File: source/geometry/FloatTransform2D.ooc (hunk 2 above)
Comment (thomasfanell) [inline source/geometry/FloatTransform2D.ooc]: I'll look into it

File: source/geometry/IntBox2D.ooc (hunk 3 above)
Comment (thomasfanell) [inline source/geometry/IntBox2D.ooc]: You can probably do that, yeah... never overloaded `as` myself, but i've seen it around.

File: source/geometry/Quaternion.ooc [hunk 4]
```diff
@@ -177,36 +108,54 @@ Quaternion: cover {
 			result
 		}
 	}
```
Comment (marcusnaslund) [inline source/geometry/Quaternion.ooc]: Just as a note, this should be
```ooc
exponential: This { get {
  ...
}}```

**Comment by marcusnaslund** [comment]
I assume you'll change the ordering of the operators in a later PR?

File: source/geometry/Quaternion.ooc (hunk 4 above)
Comment (thomasfanell) [inline source/geometry/Quaternion.ooc]: not `exponential: This { get { result }}` ?

**Comment by thomasfanell** [comment]
Yep

File: source/geometry/Quaternion.ooc (hunk 4 above)
Comment (marcusnaslund) [inline source/geometry/Quaternion.ooc]: It's eight lines. :)

Comment (thomasfanell) [inline source/geometry/Quaternion.ooc]: Oh, so a one-liner is not acceptable then, I take it... ;-)

Comment (marcusnaslund) [inline source/geometry/Quaternion.ooc]: If you can do it without semicolons, sure :D

**Pull request merged**